controllers/auth.controller.js

```diff
@@ -1,389 +1,387 @@
+    email: req.body.email
+  })
+
+  if (existingUser) {
+    return next(
+      new AppError('User already exist, please use the login route.', 400)
+    )
+  }
+
+  // hashPassword
+  const hashedPassword = await hashPassword(req.body.password, 10)
+
+  // create a new user
+  const newUser = new User({
+    firstName: req.body.firstName,
+    lastName: req.body.lastName,
+    email: req.body.email,
+    password: hashedPassword
+  })
+  const accessToken = jwt.sign(
+    {
+      userId: newUser._id
+    },
+    process.env.JWT_SECRET,
+    {
+      expiresIn: '1h'
     }
-    
-    // check if user exists
-    const existingAdmin = await Admin.findOne({
-        email: req.body.email,
-    });
-     
-    if (existingAdmin) {
-        return next(
-            new AppError(
-                'Admin already exist, please use the login route.',
-                400
-            )
-        );
-    };
-     
-    // hashPassword
-    const hashedPassword = await hashPassword(req.body.password, 10);
-      
-    try {
-         
-     // create a new user
+  )
+  newUser.accessToken = accessToken
+  await newUser.save()
+
+  return res.status(201).json({
+    status: 'success',
+    message: 'Registration successful',
+    user: newUser
+  })
+})
+
+exports.adminSignup = asyncWrapper(async (req, res, next) => {
+  // validate user body request
+  const { error } = validateSignUp(req.body)
+  if (error) {
+    console.log(error)
+    return next(new AppError(error.message, 400))
+  }
+
+  // check if user exists
+  const existingAdmin = await Admin.findOne({
+    email: req.body.email
+  })
+
+  if (existingAdmin) {
+    return next(
+      new AppError('Admin already exist, please use the login route.', 400)
+    )
+  }
+
+  // hashPassword
+  const hashedPassword = await hashPassword(req.body.password, 10)
+
+  try {
+    // create a new user
     const newAdmin = new Admin({
-         firstName: req.body.firstName,
-         lastName: req.body.lastName,
-         email: req.body.email,
-         password: hashedPassword,
-         photo: {
-            storagePath: `/public/uploads/adminPhoto/${req.file.filename}`,
-            contentType: req.file.mimetype,
-        }
-     });
-      
-      
-     const accessToken = jwt.sign({ 
-         userId: newAdmin._id, userRole: newAdmin.role}, 
-         process.env.JWT_SECRET, {
-         expiresIn: "1h"
-     });
-      
-     newAdmin.accessToken = accessToken;
-     await newAdmin.save();
-     console.log("here3")
-     return res.status(201).json({
-         status: 'success',
-         message: 'Registration successful',
-         newAdmin,
-     });
-    } catch (error) {
-      res.status(500).json({
-         message: "Internal server error",
-         error: error
-      })
-    }
- });
+      firstName: req.body.firstName,
+      lastName: req.body.lastName,
+      email: req.body.email,
+      password: hashedPassword,
+      photo: {
+        storagePath: req.file.path,
```

Comment:
I purposely didn't set this to "req.file.path", the path it returns cannot be called in the frontend to access the picture (it returns a path with back slashes instead of front slashes for a normal URL). I don't know how cloudinary multer returns its path. Please console.log it and send me a screenshot.

Reply:
yes 'req.file.path' before was an absolute path. i made it relative with an actual url link, you dont need to access it from your disk storage anymore. it has an https link that can now be directly accessed from any where.